### DISCUSSION TITLE
Definition of fully supported transcript models in QC html report

### DISCUSSION THREAD

**Discussion by tranStephen** [ConesaLab/SQANTI3#357] [Q&A] [answered]
I ran sqanti3_qc.py with  --report html option set. There is a set of graphs produced in the "Supported Transcript Models" section. What is the definition of "Fully supported" versus "Not fully supported"?
![newplot](https://github.com/user-attachments/assets/0bddfad9-3f67-43aa-9746-5af404afacd7)

**Comment by lkondratova** [comment, marked as answer]
Hi @tranStephen,
Fully supported isoforms have the following characteristics:
**3' end supported**: The transcript's 3' end is located within 50 nucleotides upstream or downstream of the 3' end of the reference isoform. If polyA motif information was supplied, it also means that the 3' end of the isoform is located within a polyA motif.
**5' end supported**: The transcript's 5' end is located within 50 nucleotides upstream or downstream of the 5' end of the reference isoform. If CAGE peaks were provided, the 5' end of the isoform is located within a CAGE peak.
**Splice junctions are supported**: The isoform is FSM/ISM, or all of its splice junctions are supported by short reads.

**Reply by qgong13** [reply]
Hello, I have the same confusion and would like to clarify that, when it is categorized as "not fully supported" does it mean that the splice junction are not fully supported? If so why would certain transcript be categorized as "not fully supported FSM/ISM" instead of being categorize in to the NIC/NNC? And what about those "not fully supported NIC/NNC"?

**Reply by TianYuan-Liu** [reply]
That’s a great question. For FSMs, only reference matches (i.e., complete agreement at both 5′ and 3′ ends) are considered “fully supported.” Cases labeled as “not fully supported” usually arise due to variation at the transcript boundaries (5′ or 3′ ends), while all internal splice junctions are still supported.

**Reply by qgong13** [reply]
Thanks for the quick response @TianYuan-Liu ! 

However your explanation does not fully clear my confusion. I understand that when it is defined as "alternative", the difference at the corresponding end is >50 bp, however I am seeing sub-category as "not fully supported alternative 3' and/or 5' end" in my qc report as well, which does not seem to align with the definition  in your explanation:
<img width="673" height="352" alt="image" src="https://github.com/user-attachments/assets/b83553d8-ec72-42bd-bb8f-5dc8f5f3aee3" />

In addition, could you also explain the context of "fully supported" vs "not fully supported" in the two novel categories, NIC and NNC? 

Thanks!

**Reply by TianYuan-Liu** [reply]
Sorry for the late reply @qgong13 , and thanks for catching that!

The distinction between “fully supported” and “not fully supported” mainly comes from the TSS/TTS distances — specifically how close the transcript ends are to the annotated sites for both novel and annotated transcripts.

The misalignment you noticed in the QC report was actually due to an earlier mistake in the error report generation logic. This has already been corrected in the latest master branch. If you pull the most recent update, the classification should now be consistent with the definitions.

We didn’t push a new release for this since it was a minor fix, but the master branch should give you the correct behavior.

Thanks again for pointing this out!
